backend/src/apiserver/resource/resource_manager.go

```diff
@@ -195,616 +510,1147 @@ func (r *ResourceManager) ArchiveExperiment(ctx context.Context, experimentId st
 			opts, err = list.NewOptionsFromToken(newToken, 50)
 			if err != nil {
 				return util.NewInternalServerError(err,
-					"Failed to create list jobs options from page token when archiving experiment. ")
+					"Failed to create list jobs options from page token when archiving experiment %v", experimentId)
 			}
 		}
 	}
 	return r.experimentStore.ArchiveExperiment(experimentId)
 }
 
+// Un-archives the experiment with the given id.
 func (r *ResourceManager) UnarchiveExperiment(experimentId string) error {
 	return r.experimentStore.UnarchiveExperiment(experimentId)
 }
 
-func (r *ResourceManager) ListPipelines(filterContext *common.FilterContext, opts *list.Options) (
-	pipelines []*model.Pipeline, total_size int, nextPageToken string, err error) {
-	return r.pipelineStore.ListPipelines(filterContext, opts)
-}
-
-func (r *ResourceManager) GetPipeline(pipelineId string) (*model.Pipeline, error) {
-	return r.pipelineStore.GetPipeline(pipelineId)
+// Deletes the experiment with the given id.
+func (r *ResourceManager) DeleteExperiment(experimentId string) error {
+	_, err := r.experimentStore.GetExperiment(experimentId)
+	if err != nil {
+		return util.Wrapf(err, "Failed to delete experiment %v due to error fetching it", experimentId)
+	}
+	return r.experimentStore.DeleteExperiment(experimentId)
 }
 
-func (r *ResourceManager) GetPipelineByNameAndNamespace(name string, namespace string) (*model.Pipeline, error) {
-	return r.pipelineStore.GetPipelineByNameAndNamespace(name, namespace)
-}
+// Creates a pipeline, but does not create a pipeline version.
+// Call CreatePipelineVersion to create a pipeline version.
+func (r *ResourceManager) CreatePipeline(p *model.Pipeline) (*model.Pipeline, error) {
+	// Assign the default namespace if it is empty
+	if p.Namespace == "" {
+		p.Namespace = r.defaultNamespace
+	}
 
-func (r *ResourceManager) DeletePipeline(pipelineId string) error {
-	_, err := r.pipelineStore.GetPipeline(pipelineId)
+	// Create a record in KFP DB (only pipelines table)
+	newPipeline, err := r.pipelineStore.CreatePipeline(p)
 	if err != nil {
-		return util.Wrap(err, "Delete pipeline failed")
+		return nil, util.Wrap(err, "Failed to create a pipeline in PipelineStore")
 	}
 
-	// Mark pipeline as deleting so it's not visible to user.
-	err = r.pipelineStore.UpdatePipelineStatus(pipelineId, model.PipelineDeleting)
+	newPipeline.Status = model.PipelineReady
+	err = r.pipelineStore.UpdatePipelineStatus(
```

Comment:
By looking at the code in master, the original `CreatePipeline` has logic to store the pipeline file to a path dependent on pipeline version. Is this logic preserved?

Reply:
Yes. We preserved this in this PR. However, as we have introduced the column `PipelineSpec` in the database, we may opt out for that option later.